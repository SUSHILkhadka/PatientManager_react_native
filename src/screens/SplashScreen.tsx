import React from 'react';
import {
  ActivityIndicator,
  Dimensions,
  SafeAreaView,
  StyleSheet,
  Text,
  View,
} from 'react-native';
import {COLOR} from '../components/styles/constants';
import {safeAreaStyles} from './login/LoginPage';

const SplashScreen = () => {
  return (
    <SafeAreaView testID="main" style={safeAreaStyles.page}>
      <View style={splashScreenStyle.textContainer}>
        <Text testID="ff" style={splashScreenStyle.title}>
          Patient PM
        </Text>
        <ActivityIndicator size={40} color={COLOR.pink2} />
      </View>
    </SafeAreaView>
  );
};

const splashScreenStyle = StyleSheet.create({
  textContainer: {
    width: '100%',
    height: Dimensions.get('window').height,
    backgroundColor: COLOR.black1,
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    justifyContent: 'center',
  },
  title: {
    color: COLOR.pink1,
    fontSize: 40,
    fontWeight: 'bold',
  },
});
export default SplashScreen;
