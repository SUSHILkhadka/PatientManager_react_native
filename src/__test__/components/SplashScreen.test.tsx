import React from 'react';
import {render} from '@testing-library/react-native';
import SplashScreen from '../../screens/SplashScreen';

it('should render splash screen properly', () => {
  const page = render(<SplashScreen />);
  expect(page).toMatchSnapshot();
});
