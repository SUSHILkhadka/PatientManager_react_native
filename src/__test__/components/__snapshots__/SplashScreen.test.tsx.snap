// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render splash screen properly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#101010",
      "flex": 1,
    }
  }
  testID="main"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#101010",
        "display": "flex",
        "flexDirection": "column",
        "height": 1334,
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#ff2394",
          "fontSize": 40,
          "fontWeight": "bold",
        }
      }
      testID="ff"
    >
      Patient PM
    </Text>
    <ActivityIndicator
      color="#e51c83"
      size={40}
    />
  </View>
</RCTSafeAreaView>
`;
